### PR TITLE
[feature-wip](unique-key-merge-on-write) unique key with merge on write table support schema change

### PR DESCRIPTION
**Pull request by liaoxin01**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/schema_change.cpp [hunk 1]
```diff
@@ -1938,20 +1958,86 @@ Status SchemaChangeHandler::_do_process_alter_tablet_v2(const TAlterTabletReqV2&
...
+            int64_t max_version = new_tablet->max_version().second;
+            std::vector<RowsetSharedPtr> rowsets;
+            if (end_version < max_version) {
+                LOG(INFO)
+                        << "alter table for unique with merge-on-write, calculate delete bitmap of "
+                        << "double write rowsets for version: " << end_version + 1 << "-"
+                        << max_version;
+                RETURN_IF_ERROR(new_tablet->capture_consistent_rowsets(
+                        {end_version + 1, max_version}, &rowsets));
+            }
```
Comment (zhannngchen) [inline be/src/olap/schema_change.cpp:1995]: Add a comment here, it's step2

File: be/src/olap/schema_change.cpp [hunk 2]
```diff
@@ -1938,20 +1958,86 @@ Status SchemaChangeHandler::_do_process_alter_tablet_v2(const TAlterTabletReqV2&
...
+            for (auto rowset_ptr : rowsets) {
+                if (rowset_ptr->version().second <= end_version) {
+                    continue;
+                }
+                std::lock_guard<std::mutex> rwlock(new_tablet->get_rowset_update_lock());
+                std::shared_lock<std::shared_mutex> wrlock(new_tablet->get_header_lock());
+                RETURN_IF_ERROR(new_tablet->update_delete_bitmap_without_lock(rowset_ptr));
+            }
+
```
Comment (zhannngchen) [inline be/src/olap/schema_change.cpp:2004]: // step3

File: be/src/olap/schema_change.cpp [hunk 3]
```diff
@@ -1938,20 +1958,86 @@ Status SchemaChangeHandler::_do_process_alter_tablet_v2(const TAlterTabletReqV2&
...
+                        << new_max_version;
+                RETURN_IF_ERROR(new_tablet->capture_consistent_rowsets(
+                        {max_version + 1, new_max_version}, &rowsets));
+            }
+            for (auto rowset_ptr : rowsets) {
+                if (rowset_ptr->version().second <= max_version) {
+                    continue;
+                }
+                RETURN_IF_ERROR(new_tablet->update_delete_bitmap_without_lock(rowset_ptr));
+            }
```
Comment (zhannngchen) [inline be/src/olap/schema_change.cpp:2022]: // step4

**Comment by zhannngchen** [comment]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**